src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityContext.cs

```diff
@@ -84,9 +85,29 @@ internal sealed class ImmutabilityContext {
 				return true;
 			}
 
-			if( TryGetImmutableTypeInfo( type, out ImmutableTypeInfo info ) ) {
-				if( info.Kind.HasFlag( kind ) ) {
-					return true;
+			if( type is INamedTypeSymbol namedType ) {

```

Comment:
I don't thiiiink this is unreasonable. But tests will confirm down the road (this might break analyzing the declaration of an immutable container, when trying to store `T`)